icekit/contrib/navigation/abstract_models.py

```diff
@@ -0,0 +1,44 @@
+from django.db import models
+from django.utils.six import python_2_unicode_compatible
+from django.utils.translation import ugettext_lazy as _
+from fluent_contents.models import PlaceholderField, ContentItem
+from icekit.fields import ICEkitURLField
+
+
+@python_2_unicode_compatible
+class AbstractNavigation(models.Model):
+    name = models.CharField(max_length=255)
+    slug = models.SlugField()
+    pre_html = models.TextField(blank=True, default='<nav><ul>')
+    post_html = models.TextField(blank=True, default='</ul></nav>')
+    content = PlaceholderField('navigation_content')
+
+    class Meta:
+        abstract = True
+
+    def __str__(self):
+        return self.name
+
+
+@python_2_unicode_compatible
+class AbstractNavigationItem(ContentItem):
+    title = models.CharField(max_length=255)
+    url = ICEkitURLField()
```

Comment:
Perhaps access this through a `get_absolute_url` method - we may want plugins in future that don't have this field but e.g. link straight a model, or otherwise generate their URL indirectly?